src/modelfree/scheduling.py

```diff
@@ -44,8 +68,16 @@ def get_func(self, func_type):
         :param func_type: str, must be in self.allowed_func_types
         :return: callable which takes one optional argument - frac_remaining
         """
-        assert func_type in self.allowed_func_types
-        return functools.partial(self.get_val, func_type)
+        self._validate_func_type(func_type)
+        if self.func_dict.get(func_type) is None:
```

Comment:
More idiomatic: `if func_type in self.func_dict`

Reply:
Not equivalent in this case. In particular i want to capture the condition `if func_type not in self.func_dict or self.func_dict[func_type] is None:`